app/scripts/controllers/transactions/index.js

```diff
@@ -2071,93 +2004,6 @@ export default class TransactionController extends EventEmitter {
     });
   }
 
-  _calculateTransactionsCost(txMeta, approvalTxMeta) {
-    let approvalGasCost = '0x0';
-    if (approvalTxMeta?.txReceipt) {
-      approvalGasCost = calcGasTotal(
-        approvalTxMeta.txReceipt.gasUsed,
-        approvalTxMeta.txReceipt.effectiveGasPrice,
-      );
-    }
-    const tradeGasCost = calcGasTotal(
-      txMeta.txReceipt.gasUsed,
-      txMeta.txReceipt.effectiveGasPrice,
-    );
-    const tradeAndApprovalGasCost = new BigNumber(tradeGasCost, 16)
-      .plus(approvalGasCost, 16)
-      .toString(16);
-    return {
-      approvalGasCostInEth: Number(hexWEIToDecETH(approvalGasCost)),
-      tradeGasCostInEth: Number(hexWEIToDecETH(tradeGasCost)),
-      tradeAndApprovalGasCostInEth: Number(
-        hexWEIToDecETH(tradeAndApprovalGasCost),
-      ),
-    };
-  }
-
-  _trackSwapsMetrics(txMeta, approvalTxMeta) {
```

Comment:
This function is specific to swaps, but in deep it triggers `trackEvent` so I put it under `metrics.ts` (`onPostTransactionBalanceUpdate`)